### PR TITLE
[DUOS-2974] Refactor the Duos Header Component

### PR DESCRIPTION
**Pull request by PintoGideon**
### Addresses

https://broadworkbench.atlassian.net/jira/software/c/projects/DUOS/boards/123?selectedIssue=DUOS-2974

### Summary

- Remove the usage of react-hyperscript-helpers
- Convert from class component in DuosHeader to function component
- Rename the file extension from js to jsx
----
Have you read [Terra's Contributing Guide](https://github.com/DataBiosphere/terra-ui/wiki/Contributor-Guide) lately? If not, do that first.

- Label PR with a Jira ticket number and include a link to the ticket
- Label PR with a security risk modifier [no, low, medium, high]
- PR describes scope of changes
- Get a minimum of one thumbs worth of review, preferably two if enough team members are available
- Get PO sign-off for all non-trivial UI or workflow changes
- Verify all tests go green
- Test this change deployed correctly and works on dev environment after deployment

### CODE REVIEW

File: src/components/DuosHeader.jsx [hunk 1]
```diff
@@ -0,0 +1,537 @@
...
+    <SupportRequestModal
+      showModal={state.showSupportRequestModal}
+      onOkRequest={okSupportRequestModal}
+      onCloseRequest={closeSupportRequestModal}
+      url={props.location.pathname}
+    />
+  );
+
+  const tabs = headerTabsConfig.filter((data) => data.isRendered(currentUser));
+
```
Comment (aarohinadkarni) [inline src/components/DuosHeader.jsx:428]: I think we are missing some comments from the previous version:
```suggestion
  // returns true if the current page the app is on is a part of this tab
```

File: src/components/DuosHeader.jsx [hunk 2]
```diff
@@ -0,0 +1,537 @@
...
+  // note: location.state.selectedMenuTab will be populated if the user navigated
+  // to the current page by clicking on a tab from the nav bar.
+
+  // populate initialTab based on state (if valid) or by manually searching through all tabs.
+  if (location?.state?.selectedMenuTab && tabs.length > location.state.selectedMenuTab && isValidTab(tabs[location.state.selectedMenuTab])) {
+    initialTab = location.state.selectedMenuTab;
+  } else {
+    initialTab = tabs.findIndex(isValidTab);
+  }
+
```
Comment (aarohinadkarni) [inline src/components/DuosHeader.jsx:454]: Same here:
```suggestion
  // populate initialSubTab
```

File: src/components/DuosHeader.jsx [hunk 3]
```diff
@@ -0,0 +1,537 @@
...
+        return subtab.link === location.pathname || location.pathname.includes(subtab.search);
+      });
+    }
+  }
+
+  return (
+    <nav className="navbar-duos" role="navigation">
+      <Hidden mdDown={true}>
+        <div className="row no-margin" style={{ width: '100%' }}>
+          <NavigationTabsComponent
```
Comment (aarohinadkarni) [inline src/components/DuosHeader.jsx:471]: This comment should appear between lines 463 & 464:
`          //Standard navbar for medium sized displays and higher (pre-existing navbar) `

**Review by aarohinadkarni — APPROVED** [review]
looks good besides a few comments to add back in 👍

File: src/components/DuosHeader.jsx [hunk 4]
```diff
@@ -0,0 +1,540 @@
...
+    setSelectedSubTab(initialSubTab === -1 ? false : initialSubTab);
+  }, [initialTab, initialSubTab]);
+
+  return (
+    <div className={`navbar-logged ${orientation === 'vertical' ? 'navbar-vertical' : ''}`}>
+      {makeNotifications()}
+      <ul className="navbar-main">
+        <div style={{ width: '100%', display: 'flex', justifyContent: 'flex-start', alignItems: 'center' }}>
+          <Link
+            isRendered={orientation === 'horizontal'}
```
Comment (rushtong) [inline src/components/DuosHeader.jsx]: `isRendered` is not a property on `Link`

File: src/components/DuosHeader.jsx [hunk 5]
```diff
@@ -0,0 +1,540 @@
...
+                  </a>
+                </li>
+                {contactUsButton}
+                {supportrequestModal}
+              </ul>
+            )
+          };
+        </div>
+        {isLogged && (
+          <div
```
Comment (rushtong) [inline src/components/DuosHeader.jsx:230]: Can we add back any missing comments?
```suggestion
           // Navbar right side
           <div
```

File: src/components/DuosHeader.jsx [hunk 6]
```diff
@@ -0,0 +1,540 @@
...
+                  <a id="link_signOut" onClick={signOut}>Sign out</a>
+                </li>
+              </ul>
+            </li>
+          </div>
+        )}
+      </ul>
+
+      {tabs[selectedMenuTab]?.children && (
+        <Box className="duos-navigation-box navbar-sub">
```
Comment (rushtong) [inline src/components/DuosHeader.jsx:260]: nit: add back comments
```suggestion
        // Sub Tabs
        <Box className="duos-navigation-box navbar-sub">
```

**Review by rushtong — COMMENTED** [review]
This looks to preserve all existing functionality. Please see minor comments inline, thank you!

File: src/components/DuosHeader.jsx (hunk 4 above)
Comment (PintoGideon) [reply src/components/DuosHeader.jsx]: Oh no, @rushtong I missed this. I'll be more careful next time. I never thought I'd miss Typescript so much.

Comment (rushtong) [reply src/components/DuosHeader.jsx]: Hah, no worries @PintoGideon - TS conversion is also one of the things we're trying to do alongside all of the other migrations we're making. I don't necessarily recommend that for current JSX work, but it is something we want to move to.

**Review by rushtong — APPROVED** [review]
Thank you for the changes! 👍🏽

**Pull request merged**